validator/node/node.go

```diff
@@ -628,6 +624,46 @@ func proposerSettings(cliCtx *cli.Context, db iface.ValidatorDB) (*validatorServ
 	return vpSettings, nil
 }
 
+func handleNoProposerSettingsFlagsProvided(cliCtx *cli.Context,
+	db iface.ValidatorDB,
+	builderConfigFromFlag *validatorServiceConfig.BuilderConfig) (*validatorServiceConfig.ProposerSettings, error) {
+	// checks db if proposer settings exist if none is provided.
+	settings, err := db.ProposerSettings(cliCtx.Context)
+	if err == nil {
+		// process any overrides to builder settings
+		overrideBuilderSettings(settings, builderConfigFromFlag)
+		// if settings are empty
+		return settings, nil
+	}
+
+	if cliCtx.Bool(flags.EnableBuilderFlag.Name) {
+		// if there are no proposer settings provided, create a default where fee recipient is not populated, this will be skipped for validator registration on validators that don't have a fee recipient set.
+		// skip saving to DB if only builder settings are provided until a trigger like keymanager API updates with fee recipient values
+		return &validatorServiceConfig.ProposerSettings{
+			DefaultConfig: &validatorServiceConfig.ProposerOption{
+				BuilderConfig: builderConfigFromFlag,
+			},
+		}, nil
+	}
+	return nil, nil
+}
+
+func overrideBuilderSettings(settings *validatorServiceConfig.ProposerSettings, builderConfigFromFlag *validatorServiceConfig.BuilderConfig) {
+	// override the db settings with the results on whether the --enable-builder flag is provided.
+	// if the user does not which for settings saved in the db to be reset
```

Comment:
This sentence needs fixing

Reply:
not sure what happened here, just removing it.